src/elements/content-sidebar/activity-feed/activity-feed/ActivityFeed.js

```diff
@@ -55,33 +61,50 @@ class ActivityFeed extends React.Component<Props, State> {
 
     componentDidUpdate(prevProps: Props, prevState: State) {
         const { feedItems: prevFeedItems } = prevProps;
-        const { feedItems: currFeedItems } = this.props;
+        const { feedItems: currFeedItems, activeFeedEntryId } = this.props;
         const { isInputOpen: prevIsInputOpen } = prevState;
         const { isInputOpen: currIsInputOpen } = this.state;
 
-        const isEmpty = this.isEmpty(this.props);
-        const wasEmpty = this.isEmpty(prevProps);
-        const hasLoaded = isEmpty !== wasEmpty && !isEmpty;
-
+        const hasLoaded = this.hasLoaded();
         const hasMoreItems = prevFeedItems && currFeedItems && prevFeedItems.length < currFeedItems.length;
-        const hasNewItems = !prevFeedItems && currFeedItems;
+        const didLoadFeedItems = prevFeedItems === undefined && currFeedItems !== undefined;
         const hasInputOpened = currIsInputOpen !== prevIsInputOpen;
 
-        if (hasLoaded || hasMoreItems || hasNewItems || hasInputOpened) {
+        if ((hasLoaded || hasMoreItems || didLoadFeedItems || hasInputOpened) && activeFeedEntryId === undefined) {
             this.resetFeedScroll();
         }
+
+        // do the scroll only once after first fetch of feed items
+        if (didLoadFeedItems) {
+            this.scrollToActiveFeedItem();
+        }
+    }
+
+    scrollToActiveFeedItem() {
+        const { activeFeedEntryId } = this.props;
+        if (!activeFeedEntryId) {
+            return;
+        }
+        if (activeFeedEntryId && this.activeFeedItemRef.current) {
```

Comment:
Is this condition duplicated in the guard clause? Is it needed or can we check the existence of the ref itself? Can we destructure the ref variable, such as:

```
const { current: activeFeedItemRef } = this.activeFeedItemRef;

if (!activeFeedItemRef) {
    return;
}

scrollIntoView(activeFeedItemRef);
```